arch/arm/soc/nordic_nrf5/nrf52/Makefile

```diff
@@ -10,6 +10,7 @@ soc-cflags += -DNRF52840_XXAA
 endif
 
 obj-y += soc.o
+obj-$(CONFIG_NRF52_ARM_MPU_ENABLE) += arm_mpu_regions.o
```

Comment:
Both the ARM Beetle and STM32 implementation used this naming structure.  You have a good point however.